### ISSUE TITLE
Technical question about Line Items API

### ISSUE THREAD

**Issue by roycornelissen** [skarpdev/dotnetcore-hubspot-client#47]
Hi,

This is more of a technical question...

I have a few more entities I need to support: `Product` and `Line Item` (well, I need at least the latter because I need to create line items from our application), so I'm attempting to add them to this library. Mind you I'm quite new to the HubSpot API so it takes some research for me to figure it out... I think I can cover most of the operations and map them to the structure of this component. I just wonder how to map the result from this method: https://legacydocs.hubspot.com/docs/methods/line-items/batch-get-line-items

```json
{
  "9845651": {
    "objectType": "LINE_ITEM",
    "portalId": 62515,
    "objectId": 9845651,
    "properties": {
      "hs_product_id": {
        "versions": [
          {
            "name": "hs_product_id",
            "value": "1645342",
            "sourceVid": []
          }
        ],
        "value": "1645342",
        "timestamp": 0,
        "source": null,
        "sourceId": null
      }
    },
    "version": 1,
    "isDeleted": false
  },
  "9867373": {
    "objectType": "LINE_ITEM",
    "portalId": 62515,
    "objectId": 9867373,
    "properties": {
      "hs_product_id": {
        "versions": [
          {
            "name": "hs_product_id",
            "value": "1645187",
            "sourceVid": []
          }
        ],
        "value": "1645187",
        "timestamp": 0,
        "source": null,
        "sourceId": null
      }
    },
    "version": 1,
    "isDeleted": false
  }
}
```

This seems like a dictionary, and doesn't really look like the other response types you use in this library, am I correct?

**Comment by nover** [comment]
Oh my what a response - I actually don't use HubSpot anymore but I have left the client library online as I can see that the downloads keep coming in on nuget.

But the response do indeed look different than all the previous mappings done, to some extend at least. The `properties` part look familiar.

I guess a custom (de)serialiser will be needed here to support the response data.

**Comment by roycornelissen** [comment]
It solves my need for my current project quite well so I'm happy it's here :) I especially like that it's .NET Core. Tried an alternative but that doesn't work well with .NET Core and is not async.

I'll see what I can do with this. The other methods look quite standard, so I may postpone this `batch-get-line-items`. I mostly need to be able to create new ones 😬

**Comment by nover** [comment]
Yes I also ended up creating the library because the landscape back in 2017 for a dotnet core HubSpot client was not great.

I would appreciate it if you give it a stab with the line items, my work-life is quite busy at the moment so I will have a hard time finding slots to sit down and figure it out.

I guess a lot of the logic from the `ReqestDataConverter` can be used / extended to solve this deserialiser problem:
https://github.com/skarpdev/dotnetcore-hubspot-client/blob/master/src/Core/Requests/RequestDataConverter.cs

**Comment by roycornelissen** [comment]
👍 Totally understand, I need it for my project anyway so I can spend work time on this right now 😄  

I'm working on the line items. The API is a bit "exotic" as in the request and response types are not very symmetric for the batch operations 😒

**Comment by roycornelissen** [comment]
Since the basic LineItems operations are now implemented, I think this issue can be closed. The `batch get line items` is not part of the implementation though.

**Comment by nover** [comment]
@roycornelissen closed and opened new issue to track it #54
